gui/src/2D/controls/multiLine.ts

```diff
@@ -205,8 +207,13 @@ export class MultiLine extends Control {
                 first = false;
             }
             else {
-                context.lineTo(point._point.x, point._point.y);
+                if (point._point.z < 1 && previousPoint.z < 1) {

```

Comment:
Shouldn't you also test that z is >= 0?

If the point is behind the camera, z will be < 0 (I think z is the screen depth?)

Also, I think that when a point is visible but not the other, the line will disappear completely? [...] I have just tested, it's what's happening.

I guess the right way would be to clip the line "by hand" in homogenous space against the view frustum...

Reply:
nop because we are in clip space and not in view space.
Yes, the right way to do it is to clip the lines in world space with the near plane. But I don't want to go down that route 2 weeks before release as it needs a bigger refactor.

Reply:
I tried to lerp with the plan Z to get clipped lines. it almost work but I'm not sure I can interpolate with clipspace Z. I prefer to let it that way and see if it's an issue later.